### DISCUSSION TITLE
Copy Instance

### DISCUSSION THREAD

**Discussion by zax71** [Minestom/Minestom#1979] [Help] [answered]
How can I make a copy of an instance? On the Discord there is reference to `Instance#copy()` but that doesn't seam to exist? A code example of how I am obtaining the "master" instance is below

```java
Instance selectedMap = SpleefMinigame.spleefMinigame.getInstances().stream()
                .filter((map) -> map.getTag(Tag.String("name")).equals(selectedMapName))
                .findFirst()
                .orElse(null);

// Non existant method
selectedMap = selectedMap.copy();
```

Adding on to this, would it be possible to:
 1. Set an instance to be immutable (for the master instance)
 2. Explicitly remove an instance for once a game has ended, I could see these unused instances not being removed and taking up resources

**Comment by KrystilizeNevaDies** [comment]
If you use a shared instance, your children instances will share all the blocks and be immutable.

**Comment by mworzala** [comment, marked as answer]
The copy method you're referring to is on InstanceContainer. https://javadoc.hollowcube.net/minestom-ce/net/minestom/server/instance/InstanceContainer#copy()

**Reply by zax71** [reply]
Ah, I see. Thanks!

**Reply by zax71** [reply]
Is there a way to make `InstanceContainer#copy()` copy tags as well? These seam to be lost

**Reply by mworzala** [reply]
It probably should. If not I would consider it a bug, but you can always copy them manually in the mean time

**Reply by zax71** [reply]
~~I'll make a bug on CE~~ done, [#112](https://github.com/hollow-cube/minestom-ce/issues/112). Regarding copying them manually, I poked around for a while, but I cannot find a way to get all the tags associated with the instance. Hardcoding them seams like a bit of a nuisance.
